lldb/source/ValueObject/DILEval.cpp

```diff
@@ -740,16 +764,182 @@ Interpreter::Visit(const BooleanLiteralNode *node) {
   return ValueObject::CreateValueObjectFromBool(m_target, value, "result");
 }
 
+llvm::Expected<CastKind>
+Interpreter::VerifyCastType(lldb::ValueObjectSP operand, CompilerType op_type,
+                            CompilerType target_type, int location) {
+
+  if (target_type.IsScalarType()) {
```

Comment:
Can we put everything that's inside this `target_type.IsScalarType()` if-block into a helper. Call it, e.g., `VerifyArithmeticCastType` or something